src/index.ts

```diff
@@ -131,11 +146,15 @@ function getHandlers(
       const velocity = Math.sqrt(absX * absX + absY * absY) / (time || 1);
       const vxvy: Vector2 = [deltaX / (time || 1), deltaY / (time || 1)];
 
+      const dir = getDirection(absX, absY, deltaX, deltaY);
+
       // if swipe is under delta and we have not started to track a swipe: skip update
-      if (absX < props.delta && absY < props.delta && !state.swiping)
-        return state;
+      const delta =
+        typeof props.delta === "number"
+          ? props.delta
+          : props.delta[uncapitalizeDirection(dir)];
```

Comment:
💭 tried a few approaches to trim the size up a bit and avoid the `uncapitalizeDirection` method. I think we can utilize `as` here.
```suggestion
          : props.delta[dir.toLowerCase() as Uncapitalize<SwipeDirections>];
```

- [TS playground example](https://www.typescriptlang.org/play?#code/KYDwDg9gTgLgBAYwgOwM7wDIFEBiAVOAXjgCINgAzGEgbgChRJZEV04AlASQHEAJA4iXYBLAOYALavUbR4SNPACqABSKlFYWg3CyWCuABEA8gHUAcmpIGIAd2RbtTeDACeYYHADKN4e4PCoYAQYYVYiOjg4AB84V3cICjhsfAjo2LdgBI4eflSYuMzElTz0+MTjc3pHXQK4AGEUCjEAVygAQxCUIzBO5ANgABsYNvDImORmgFsAI2AoEoBvOABtAGtgFzhhZDhFZAQ2sGFhgeEAL2AAHm9fYH9A4NC0AD4AXQAuOAmZubgAXyqMmY2xgcwobQQHgayCaolaHSe3V6qDgC1SABNBsNPtDYfDekinv0hm16H86HQKM19r04NM4AAKdDNCgUHGNFrtAk9J6oAA0cHRAU+Nz8ASCyIAlKjUpFhIkGQUsszWQA6TEkuAAQkIxAA5N9ZlA9dK0ZFzeb5KgIANgKqBhBREyYCyKOqsW1lkKoKqYBAMLY5nU2qhgAzpSHdvtDsc2qcLtcfGKHsi3pLZf86H8gA)

Reply:
Even with it, it extends the size limit:
```
  Package size limit has exceeded by 114 B
  Size limit: 1.15 kB
  Size:       1.26 kB with all dependencies, minified and gzipped
  ```

Reply:
😞 that's crazy. It must be something introduced with the bundler upgrade. Quite annoying, i'll have to dig in to the outputs and see if we can figure this discrepancy out.

Also posted this typescript issue to my colleagues and they had a couple other approaches to consider that we can look at.
- _hair cleaner, but only because I find the word Uncapitalize a bit odd and unintuitive:_
  - `stuff.delta[dir.toLowerCase() as Lowercase<SwipeDirections>]`
- [nother solution](https://www.typescriptlang.org/play?module=1#code/KYDwDg9gTgLgBAYwgOwM7wDIFEBiAVOAXjgCINgAzGEgbgChRJZEV04AlASQHEAJA4iXYBLAOYALavUbR4SNPACqABSKlFYWg3CyWCuABEA8gHUAcmpIGIAd2RbtTeDACeYYHADKN4e4PCoYAQYYVYiOjg4AB84V3cICjhsfAjo2LdgBI4eflSYuMzElTz0+MTjc3o6ArgAYRQKMQBXKABDEJQjMA7kA2AAGxhWgGlgFwA1Vv6mj2IAbzgAbQBrMbhhZDhFZARWsGEh-uEAL2AAHm9fYH9A4NC0AD4AXQAuOGQmgFsAI2AoOAAvlUZMwavVkI1RC12vcuj0+oNWuFIjEPj8-iVwZDoT04fcEUNRhMpjM6GTHLoNjA-hRWggPFjmm1cd17qg4HNUgATAZDN6MqHM2GslAE1r0AFk+RsYSoDC2P61VqoYCXMA3II9dnEAAUqxcb3QUA2ogANHAIN8AFb8hpMmGdEW9XlIgCUb3163Z+qyApxwvhLqJk2msweHLgqWEFB1BSylqtREIxAA5GjflAU67OZFIoEYC1NrT+ir6JFJXngAWoJtPRsLdaJWSKE0dj04N84Dr0E0KBRbRD7SyteauQE3mrrgFNWzXRzUpFo1244ke32AHQ8xFwACEybgaa+Gaz89zuelzmAbGIY6g65gEHlNkVyuAOtdZbPuejOtlT5fKpqhqdysLGV4wOaa4UJuLqutmC5fpE0oQP0wDrv0ECiN2BYbluQyLNS6BPK6CFngCAwqqeiFIawKFoRhWEkJ6yAQPAFAQK2XJwC4EAkCR1GAghkqSnQ3w6jmcB4a0bwSZEXK2MgbwAIymghqFUG8ABMqlnsaEgwG8ADMOm5k0YBvAALKkwnmhUZgfqJ4nci6MkIfJdjKSZkTqQZcDaQhemSEZXlwGZlnWXQALmskeAOWJElSa5Z7uYpcAqWplC+f5uliEFcDGQhYVwFZ5aReaKhxU5ckuVRckKZ5GUaX5IWBb5BVnkVJWCVF2R8LFNBAA)
- [and another similar one](https://www.typescriptlang.org/play?ts=4.3.5#code/KYDwDg9gTgLgBAYwgOwM7wDIFEBiAVOAXjgCINgAzGEgbgChRJZEV04AlASQHEAJA4iXYBLAOYALavUbR4SNPACqABSKlFYWg3CyWCuABEA8gHUAcmpIGIAd2RbtTeDACeYYHADKN4e4PCoYAQYYVYiOjg4AB84V3cICjhsfAjo2LdgBI4eflSYuMzElTz0+MTjc3pHXQKk22AoAGEAQ1Rgb19gf0Dg0LQ1RWQEZrBhGGaAG2EAL2AAHg6-AKCQ1gA+KplmWsaUCjEAVyhm1eQjMFODYAnx8MiY5AOAWwAjBpKAbzgAbQBrYBccGEyDqNgaLTaiy6y16rAAugAuOCPV4NOAAX02OmYwJgDQozQQHl2yH2oiOJz651OqDgH1SABNruMkSSyRTTtS+lcbs16Oi6HR5GxhKgMPUmq12j4lj1TmoABT-FxI9BQYGiAA0cAgLwAVqy9odjpyLtzmc04ABKJHKoG08VgyWQmXQuV9WmENZ0uCpYQUBUFLK6vVEQjEADkKLeUAjVvpkUigRgRxBBImbXokQFSeAKagILtwJ1+v5gooByG8pecAV6AOFAohtJxspKC5rG1DICSKh3RWHqtdNSkX9taDiXrjYAdEzeXAAIThuBR54xuPDxOJ4XwCYStTdqDTmAQR3gqUKq1ZrdAxIK0Vn53Szr92HIBV7p3aqcUWcWq1DgmN6RMKEATMA057qIdYpjOc7jN8n4NHCVojomOboXQ6JAA)

Reply:
A +10% increase in size is not a lot but it is also sort of large when your size is already really small. I think we can find the culprit and keep the size down, will just take a bit of investigation that I can take on hopefully this week.

Reply:
@hartzis I created a diff comparting the output from this branch and from the `main`, and I got it:
```diff
2a3,24
> function _interopNamespace(e) {
>   if (e && e.__esModule) return e;
>   var n = Object.create(null);
>   if (e) {
>     Object.keys(e).forEach(function (k) {
>       if (k !== 'default') {
>         var d = Object.getOwnPropertyDescriptor(e, k);
>         Object.defineProperty(n, k, d.get ? d : {
>           enumerable: true,
>           get: function () {
>             return e[k];
>           }
>         });
>       }
>     });
>   }
>   n['default'] = e;
>   return n;
> }
>
> var React__namespace = /*#__PURE__*/_interopNamespace(React);
>
114c136,137
<       var vxvy = [deltaX / (time || 1), deltaY / (time || 1)]; // if swipe is under delta and we have not started to track a swipe: skip update
---
>       var vxvy = [deltaX / (time || 1), deltaY / (time || 1)];
>       var dir = getDirection(absX, absY, deltaX, deltaY); // if swipe is under delta and we have not started to track a swipe: skip update
116,117c139,140
<       if (absX < props.delta && absY < props.delta && !state.swiping) return state;
<       var dir = getDirection(absX, absY, deltaX, deltaY);
---
>       var delta = typeof props.delta === "number" ? props.delta : props.delta[dir.toLowerCase()];
>       if (absX < delta && absY < delta && !state.swiping) return state;
280,281c303,304
<   var transientState = React.useRef(_extends({}, initialState));
<   var transientProps = React.useRef(_extends({}, defaultProps));
---
>   var transientState = React__namespace.useRef(_extends({}, initialState));
>   var transientProps = React__namespace.useRef(_extends({}, defaultProps));
284c307
<   var _React$useMemo = React.useMemo(function () {
---
>   var _React$useMemo = React__namespace.useMemo(function () {
```

In short, the new microbundle version injects the function `_interopNamespace` to be used with React calls.
It comes from Rollup, and we can remove it by disabling the option [`output.interop`](https://rollupjs.org/guide/en/#outputinterop), but I'm not sure that it's really good to disable it.

Reply:
@macabeus You rock! amazing find!

thank you for the link to the rollup option, it looks like they are going to be changing `output.interop` to default to  `auto` so i'm curious if we can try that first. If that doesn't work, maybe we roll back upgrading `microbundle` and circumvent the need for the typescript updates for now?

Reply:
Oka oka.. quick update. TLDR, moving forward with this PR as is for now.

`microbundle` does not give us access to the rollup config, [for good reason](https://github.com/developit/microbundle/issues/816), so we're unable to edit the `output.interop` value.

I'm going to make an issue [add here] that will tackle removing `microbundle` in the future and use something else, who knows what. I will probably miss the convenience of `microbundle` but this project's output is so small I think we should have finer grained control over what is output during compile to help manage its file size.

🎉 side note, i think i found a way to "counter" the increase by removing comments from the build via `tsc` option [removeComments](https://www.typescriptlang.org/tsconfig/#removeComments)